extensions/extensionservice/org.eclipse.smarthome.extensionservice.marketplace/src/main/java/org/eclipse/smarthome/extensionservice/marketplace/internal/MarketplaceExtensionService.java

```diff
@@ -0,0 +1,404 @@
+        URL url = new URL(urlString);
+        return IOUtils.toString(url);
+    }
+
+    private void postInstalledEvent(String extensionId) {
+        if (eventPublisher != null) {
+            Event event = ExtensionEventFactory.createExtensionInstalledEvent(extensionId);
+            eventPublisher.post(event);
+        }
+    }
+
+    private void postUninstalledEvent(String extensionId) {
+        if (eventPublisher != null) {
+            Event event = ExtensionEventFactory.createExtensionUninstalledEvent(extensionId);
+            eventPublisher.post(event);
+        }
+    }
+
+    private void postFailureEvent(String extensionId, String msg) {
+        if (eventPublisher != null) {
+            Event event = ExtensionEventFactory.createExtensionFailureEvent(extensionId, msg);
+            eventPublisher.post(event);
+        }
+    }
+
+    private String getExtensionId(Node node) {
+        StringBuilder sb = new StringBuilder(EXT_PREFIX);
+        switch (node.packagetypes) {
+            case MP_PACKAGETYPE_RULE_TEMPLATE:
+                sb.append(EXT_TYPE_RULE_TEMPLATE).append("-");
+                break;
+            case MP_PACKAGETYPE_BINDING:
+                sb.append(EXT_TYPE_BINDING).append("-");
+                break;
+            default:
+                return null;
+        }
+        sb.append(node.id); // todo: remove illegal chars
+        return sb.toString();
+    }
+
+    private boolean isInstalled(String extensionId) {
+        String type = getType(extensionId);
+        if (type == null) {
+            return false;
+        }
+        switch (type) {
+            case EXT_TYPE_RULE_TEMPLATE:
+                return marketplaceRuleTemplateProvider.get(extensionId) != null;
+            case EXT_TYPE_BINDING:
+                return installedBindings.containsKey(extensionId);
+            default:
+                return false;
+        }
+    }
+
+    private String getType(String extensionId) {
+        if (extensionId.startsWith(EXT_PREFIX)) {
+            String idWithoutPrefix = extensionId.substring(EXT_PREFIX.length());
+            return idWithoutPrefix.substring(0, idWithoutPrefix.lastIndexOf('-'));
+        } else {
+            return null;
+        }
+    }
+
+    private Map<String, Long> loadInstalledBindingsMap() {
+        File dataFile = bundleContext.getDataFile(BINDING_FILE);
+        if (dataFile != null && dataFile.exists()) {
+            try (FileReader reader = new FileReader(dataFile)) {
+                LineIterator lineIterator = IOUtils.lineIterator(reader);
+                Map<String, Long> map = new HashMap<>();
+                while (lineIterator.hasNext()) {
+                    String line = lineIterator.nextLine();
+                    String[] parts = line.split(";");
+                    try {
+                        map.put(parts[0], Long.valueOf(parts[1]));
+                    } catch (Exception e) {
```

Comment:
You should check for `parts.length >= 2` before instead of execute the code and catch exceptions.
So, the only thing to catch is the NumberFormatException.